src/DatePicker.js

```diff
@@ -1,41 +1,29 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import TextInput from './TextInput';
 import idgen from './idgen';
 
-class DatePicker extends React.Component {
-  constructor(props) {
-    super(props);
+const DatePicker = props => {
+  const [id] = useState(props.id || `datepicker${idgen()}`);
+  const instance = useRef(null);
 
-    this.id = props.id || `datepicker${idgen()}`;
-  }
-
-  componentDidMount() {
-    if (typeof M !== 'undefined') {
-      const { onChange } = this.props;
-      const elem = document.getElementById(this.id);
-      const options = onChange
-        ? { ...this.props.options, onSelect: onChange }
-        : this.props.options;
-
-      this.instance = M.Datepicker.init(elem, options);
-    }
-  }
+  useEffect(() => {
+    const { onChange } = props;
+    const options = onChange
+      ? { ...props.options, onSelect: onChange }
+      : props.options;
+    const elem = document.getElementById(id);
```

Comment:
How about we use a ref instead of an id here? I know in the previous implementation this is what was there, but we can take the opportunity to improve this :)

Reply:
https://github.com/react-materialize/react-materialize/pull/933 👀

Reply:
Do we need to change the 'TextInput' component so that its ref refers to parentsRef?
For example in TextInput:
  const textInputRef = props.parentRef || useRef(null);
and in the DatePicker:
  <TextInput ref={datePickerRef} />?

Or is there another solution?

Reply:
Very good point, I missed that. Normally we would use `forwardRef` but as `TextInput` is not yet stateless let's keep the `id` for now.